common/shell-local/run.go

```diff
@@ -169,17 +168,17 @@ func createFlattenedEnvVars(config *Config) (string, error) {
 	envVars["PACKER_BUILDER_TYPE"] = config.PackerBuilderType
 
 	// expose ip address variables
-	httpAddr := common.GetHTTPAddr()
-	if httpAddr != "" {
-		envVars["PACKER_HTTP_ADDR"] = httpAddr
+	httpAddr := config.generatedData["PackerHTTPAddr"]
+	if httpAddr != nil && httpAddr != "" {
```

Comment:
does comparing an `interface{}` with a `string` work here ?

Reply:
( ditto throughout )

Reply:
I think that's fine https://play.golang.org/p/MbKYzZGIGgR